src/WebJobs.Script/Workers/Rpc/Configuration/DefaultWorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    // This class resolves worker configurations by scanning the "workers" directory within the Host for worker config files.
+    internal sealed class DefaultWorkerConfigurationResolver : IWorkerConfigurationResolver
+    {
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<WorkerConfigurationResolverOptions> _workerConfigurationResolverOptions;
+
+        public DefaultWorkerConfigurationResolver(ILoggerFactory loggerFactory, IOptionsMonitor<WorkerConfigurationResolverOptions> workerConfigurationResolverOptions)
+        {
+            ArgumentNullException.ThrowIfNull(loggerFactory);
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryWorkerConfig);
+            _workerConfigurationResolverOptions = workerConfigurationResolverOptions ?? throw new ArgumentNullException(nameof(workerConfigurationResolverOptions));
+        }
+
+        public List<string> GetWorkerConfigPaths()
+        {
```

Comment:
Code in this method is taken from `AddProviders()` method of `RPCWorkerConfigFactory`

Reply:
The previous implementation does not have a dependency on `IConfiguration`. Looks like the new code relies on `WorkerConfigurationResolverOptions` and `WorkerConfigurationResolverOptionsSetup` relies on `IConfiguration`. Is this additional change intentional? If so, can we add a comment explaining why?

https://github.com/Azure/azure-functions-host/blob/de373194d431f696220bf60ae4948c82b15e2d8e/src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs#L71-L83

Reply:
The code in the constructor of `RPCWorkerConfigFactory` depends on `IConfiguration`. That code calls this method `GetDefaultWorkersDirectory()` to prepare `workersDirPath`.

Reply:
Don't know the scenario on top of my mind. The code was added in 2018 - https://github.com/Azure/azure-functions-host/issues/2947. So, the ability to overwrite default workers dir path via configuration has been available for a long time.